File: Chapter01/Java/BooleanType.java

package main.java;
/**
 * Created by willsmith on 8/11/16.
 */
public class BooleanType
{
    public static void boolDemo()
    {
        //Java
        boolean a = true;
        boolean b = false;
        boolean c = a;

        System.out.println("a: " + a);
        System.out.println("b: " + b);
        System.out.println("c: " + c);
        System.out.println("a AND b: " + (a && b));
        System.out.println("a OR b: " + (a || b));
        System.out.println("NOT a: " + !a);
        System.out.println("NOT b: " + !b);
        System.out.println("a XOR b: " + (a ^ b));
        System.out.println("(c OR b) AND a: " + ((c || b) && a));
    }
}

File: Chapter01/C#/BooleanType.cs
using System;
namespace EverydayDataStructures
{
	public class BooleanType
	{
		public BooleanType()
		{
		}

		public static void boolDemo()
		{
			//C#
			bool a = true;
			bool b = false;
			bool c = a;

			Console.WriteLine("a: {0}", a);
			Console.WriteLine("b: {0}", b);
			Console.WriteLine("c: {0}", c);

			//Operators
			Console.WriteLine("a AND b: {0}", a && b);
			Console.WriteLine("a OR b: {0}", a || b);
			Console.WriteLine("NOT a: {0}", !a);
			Console.WriteLine("NOT b: {0}", !b);
			Console.WriteLine("a XOR b: {0}", a ^ b);

			//Order of precedence & short circuiting
			Console.WriteLine("(c OR b) AND a: {0}", (c || b) && a);
		}
	}
}

